src/test/FuzzerImpl.cpp

```diff
@@ -69,6 +73,13 @@ getShortKey(PublicKey const& pk)
     return getShortKey(pk.ed25519());
 }
 
+uint8_t
+getShortKey(LedgerKey const& key)
+{
+    auto const opaque = xdr::xdr_to_opaque(key);
+    return opaque[0];
```

Comment:
> If you've never looked at https://tools.ietf.org/html/rfc4506 then this is a good opportunity to do so. Between that and our XDR files, it should be clear why this always returns 0.

Hmm, maybe I see this.  Is it that the format of variable-length opaque data begins with the length, and byte 0 is the MSB, and `LedgerKey`s are small enough that as opaque data they'll never be large enough to have a length with any bits set in the MSB?

> I explained how you can implement this correctly in an earlier comment.

I think you're referring to:

> Every LedgerKey has either a hash or a public key. Use the same technique that @MonsieurNicolas used with getShortKey and just take a byte.

Are you suggesting to switch on the type of `LedgerKey` and take byte 0 from whichever `Hash` or `PublicKey` that type of key contains?  (The only type-independent way I can think of would be to use the hash of the key, but that'd be hashing something that, as you point out, already contains some component that's been hashed, and this is a performance-critical path.)

Reply:
I'm actually referring to a comment later in that thread: https://github.com/stellar/stellar-core/pull/2904#discussion_r572971188. Specifically, "It would depend on the key type, but this is a simple switch statement." So yes, switch on the type of `LedgerKey` and take byte 0 from the relevant `Hash` or `PublicKey`.

Reply:
Done.  Also, as a note:

> Hmm, maybe I see this. Is it that the format of variable-length opaque data begins with the length, and byte 0 is the MSB, and LedgerKeys are small enough that as opaque data they'll never be large enough to have a length with any bits set in the MSB?

I wasn't right with that guess either; the `opaque_vec` there is the raw bytes of the `LedgerKey`, _not_ the "variable-length opaque data" format specified in the RFC.  So it's the MSB of the discriminant (which is also small enough that the MSB is always zero, as the length would be if we _did_ convert the `LedgerKey` into the variable-length opaque format).